tools/misc/browse_dataset.py

```diff
@@ -45,63 +39,45 @@ def parse_args():
     return args
 
 
-def retrieve_data_cfg(config_path, skip_type, cfg_options):
-    """Retrieve the dataset config file.
-
-    Args:
-        config_path (str): Path of the config file.
-        skip_type (list[str]): List of the useless pipeline to skip.
-        cfg_options (dict): dict of configs to merge from.
-    """
-
-    def skip_pipeline_steps(config):
-        config['pipeline'] = [
-            x for x in config.pipeline if x['type'] not in skip_type
-        ]
-
-    cfg = Config.fromfile(config_path)
-    if cfg_options is not None:
-        cfg.merge_from_dict(cfg_options)
-    train_data_cfg = cfg.data.train
-    while 'dataset' in train_data_cfg and train_data_cfg[
-            'type'] != 'MultiImageMixDataset':
-        train_data_cfg = train_data_cfg['dataset']
-
-    if isinstance(train_data_cfg, Sequence):
-        [skip_pipeline_steps(c) for c in train_data_cfg]
-    else:
-        skip_pipeline_steps(train_data_cfg)
-
-    return cfg
-
-
 def main():
     args = parse_args()
-    cfg = retrieve_data_cfg(args.config, args.skip_type, args.cfg_options)
+    cfg = Config.fromfile(args.config)
+    if args.cfg_options is not None:
+        cfg.merge_from_dict(args.cfg_options)
 
-    dataset = build_dataset(cfg.data.train)
+    # register all modules in mmrotate into the registries
+    register_all_modules()
 
-    progress_bar = mmcv.ProgressBar(len(dataset))
+    dataset = DATASETS.build(cfg.train_dataloader.dataset)
+
+    visualizer = VISUALIZERS.build(cfg.visualizer)
+    visualizer.dataset_meta = dataset.metainfo
 
+    progress_bar = mmcv.ProgressBar(len(dataset))
```

Comment:
`ProgressBar` is moved to `mmengine.utils`, https://github.com/open-mmlab/mmdetection/pull/8594